tests/phpunit/tests/filesystem/direct/delete.php

```diff
@@ -0,0 +1,191 @@
+		$this->assertFalse( self::$filesystem->delete( '' ) );
+	}
+
+	/**
+	 * Tests that `WP_Filesystem_Direct::delete()` deletes an empty directory.
+	 *
+	 * @ticket 57774
+	 */
+	public function test_should_delete_an_empty_directory() {
+		$dir = self::$file_structure['test_dir']['path'] . 'directory-to-delete';
+
+		$this->assertTrue(
+			mkdir( $dir ),
+			'The directory was not created.'
+		);
+
+		$this->assertTrue(
+			self::$filesystem->delete( $dir ),
+			'The directory was not deleted.'
+		);
+	}
+
+	/**
+	 * Tests that `WP_Filesystem_Direct::delete()` deletes a directory with contents.
+	 *
+	 * @ticket 57774
+	 */
+	public function test_should_delete_a_directory_with_contents() {
+		$dir     = self::$file_structure['test_dir']['path'] . 'directory-to-delete/';
+		$file    = $dir . 'file-to-delete.txt';
+		$subdir  = $dir . 'subdirectory-to-delete/';
+		$subfile = $subdir . 'subfile-to-delete.txt';
+
+		mkdir( $dir, 0755 );
+		mkdir( $subdir, 0755 );
+		touch( $file, 0644 );
+		touch( $subfile, 0644 );
+
+		$actual = self::$filesystem->delete( $dir, true );
+
+		if ( ! $actual ) {
+			unlink( $file );
+			unlink( $subfile );
+			rmdir( $subdir );
+			rmdir( $dir );
+		}
+
+		$this->assertTrue( $actual, 'The directory was not deleted.' );
+	}
+
+	/**
+	 * Tests that `WP_Filesystem_Direct::delete()` deletes a file.
+	 *
+	 * @ticket 57774
+	 */
+	public function test_should_delete_a_file() {
+		$file = self::$file_structure['test_dir']['path'] . 'file-to-delete.txt';
+
+		touch( $file );
+
+		$this->assertTrue( self::$filesystem->delete( $file ), 'The directory was not deleted.' );
+	}
+
+	/**
+	 * Tests that `WP_Filesystem_Direct::delete()`
+	 * returns true when deleting a path that does not exist.
+	 *
+	 * @ticket 57774
+	 *
+	 * @dataProvider data_paths_that_do_not_exist
+	 *
+	 * @param string $path The path.
+	 */
+	public function test_should_return_true_when_deleting_path_that_does_not_exist( $path ) {
+		if (
+			'' === $path
+			|| str_starts_with( $path, '.' )
+			|| str_starts_with( $path, '/' )
+		) {
+			$this->markTestSkipped( 'Dangerous delete path.' );
+		}
+
+		$this->assertTrue( self::$filesystem->delete( self::$file_structure['test_dir']['path'] . $path ) );
+	}
+
+	/**
+	 * Tests that `WP_Filesystem_Direct::delete()`
+	 * returns false when a directory's contents cannot be deleted.
+	 *
+	 * @ticket 57774
+	 */
+	public function test_should_return_false_when_contents_cannot_be_deleted() {
+		global $wp_filesystem;
+
+		$wp_filesystem = new WP_Filesystem_Direct( array() );
+
+		$path = self::$file_structure['test_dir']['path'] . 'dir-to-delete/';
+
+		if ( ! is_dir( $path ) ) {
+			mkdir( $path );
+		}
+
+		// Set up mock filesystem.
+		$filesystem_mock = $this->getMockBuilder( 'WP_Filesystem_Direct' )
+								->setConstructorArgs( array( null ) )
+								// Note: setMethods() is deprecated in PHPUnit 9, but still supported.
+								->setMethods( array( 'dirlist' ) )
```

Comment:
Note:
- `setMethods()` is deprecated in PHPUnit 9 and removed in PHPUnit 10.
- The suitable replacement for this case and others in this PR is `onlyMethods()`, introduced in PHPUnit 8.3.0.
- However, as the WP Core test suite runs in PHP 5.6, this uses an earlier version of PHPUnit, so `setMethods()` must be used.
- It may be preferable to add a PHP or PHPUnit version check here and run the appropriate method, so that compatibility with PHPUnit 10+ can be achieved without being coupled to the minimum WP Core version run in the test suite, or this could be done later to address all uses of `setMethods()` in the test suite in one go.

Reply:
Related: https://github.com/Yoast/PHPUnit-Polyfills/issues/58